verilog/preprocessor/verilog_preprocess.h

```diff
@@ -88,7 +89,26 @@ class VerilogPreprocess {
   using MacroParameterInfo = verible::MacroParameterInfo;
 
  public:
-  VerilogPreprocess() {}
+  struct Config {
+    // Filter out non-matching `ifdef and `ifndef branches depending on
+    // which defines are set.
+    //
+    // This option is useful for syntax check and possibly linting in
+    // case parsing fails with all consecutive branches emitted.
+    //
+    // Note, for formatting, we do _not_ want to filter the output as we
+    // want to emit all tokens.
+    bool filter_branches = false;
```

Comment:
Maybe later, this could be upgraded to an `enum` of various branch-handling policies.

Reply:
Yeah, once the exact use will show. For the 'fallback' situation, I actually want to implement that in the analyzer itself (like it already attempts to deal with various dialects).